src/v/storage/kvstore.cc

```diff
@@ -252,12 +254,14 @@ ss::future<> kvstore::flush_and_apply_ops() {
      */
     return _segment->append(std::move(batch))
       .then([this](append_result) { return _segment->flush(); })
-      .then([this, last_offset, ops = std::move(ops)]() mutable {
+      .then([this]() { return _db_mut.get_units(); })
+      .then([this, last_offset, ops = std::move(ops)](auto units) mutable {
           for (auto& op : ops) {
-              apply_op(std::move(op.key), std::move(op.value));
+              apply_op(std::move(op.key), std::move(op.value), units);
               op.done.set_value();
           }
           _next_offset = last_offset + model::offset(1);
+          units.return_all();
```

Comment:
should this be in a `ss::defer`?

Reply:
Ah ok, all good. Thought it was needed (hence also my other comment). This looks fine to me then.